docs on how to use SLURM (#19)
nspope/singer-snakemake: SLURM.md

# Running the pipeline with SLURM

Snakemake supports using [SLURM](https://github.com/SchedMD/slurm) to distribute jobs across HPC cluster nodes.

> To run the pipeline on your SLURM cluster you will need:

Reviewer: ```suggestion
To run the pipeline on your SLURM cluster you will need to:
```

1. Install Snakemake's SLURM executor plugin
2. Create a SLURM cluster profile
3. Tell Snakemake to use SLURM

*Note*: This assumes Snakemake 8+

## Installing SLURM executor plugin

To install the SLURM executor plugin, Snakemake [recommends](https://snakemake.github.io/snakemake-plugin-catalog/plugins/executor/slurm.html) you use `pip` within your `singer-snakemake-env` conda environment:

```bash
conda activate singer-snakemake-env
pip install snakemake-executor-plugin-slurm
```

## Creating a simple SLURM profile

You can either create a __global__ or __workflow__-specific profile in Snakemake (see [here](https://snakemake.readthedocs.io/en/stable/executing/cli.html#profiles) for more details).

> We recommend you to start with a simple profile that will set shared options and resources for all steps in the pipeline.

Reviewer: ```suggestion
Start with a simple profile that will set shared options and resources for all steps in the pipeline, then adjust per rule as needed.
```

First, create a subdirectory within `singer-snakemake` called `simple_profile` where your configuration file will be placed:

```bash
mkdir simple_profile
touch config.yaml
```

The profile configuration will depend on your HPC, the exact parameters in the configuration file (e.g., chunk size), 
and the amount of data used in inference (sample size, genome size, etc.). 

> We found that for 100 human genomes split into 1Mb chunks, you'd need the following resources (to be added to your `config.yaml`):

Reviewer: ```suggestion
For example, for 100 human genomes split into 1Mb chunks, the following resources seem to be sufficient (to be added to your `config.yaml`):
```
